portfolio/src/main/java/com/google/sps/servlets/LandingDataServlet.java

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.classes.LandingSight;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Returns landing sight data as a JSON array, e.g. [{"lat": 38.4404675, "lng": -122.7144313, "missionNum": "13", "description": "landed on Feb. 32 2022"}] */
+@WebServlet("/landing-data")
+public class LandingDataServlet extends HttpServlet {
+
+  private Collection<LandingSight> landingSights;
+
+  @Override
+  public void init() {
+    landingSights = new ArrayList<>();
+
```

Comment:
Since the Scanner in Java implements the AutoClosable interface, you can use the Java try-with-resource block that will make sure your scanner is closed at the end of the block: https://www.baeldung.com/java-try-with-resources 

C# has the same feature, so it's worth trying out!